### ISSUE TITLE
Wrong IsConstructor, IsEffectivelySameAs results for constructors in compiled code

### ISSUE THREAD

**Issue by auduchinok** [fsharp/FSharp.Compiler.Service#752]
In example below `mfv` is a `MemberOrFunctionOrValue` in a type in a compiled assembly.

```csharp
if (mfv.IsConstructor) {
  var entity = mfv.EnclosingEntity;
  var hasSameMember =
    entity.MembersFunctionsAndValues.Any(m => m.IsEffectivelySameAs(mfv));
  var hasCtor =
    entity.MembersFunctionsAndValues.Any(m => m.IsConstructor);
}
```
Both `hasSameMember` and `hasCtor` are `false` while expected to be `true`.

I'm using FSharp.Compiler.Service 12.0.2.

**Comment by dsyme** [comment]
@auduchinok Any chance you could send a PR adding a failing test? With the repro above we won't be certain we've solved the problem.

**Comment by auduchinok** [comment]
@dsyme Sure! I'll try to write a proper test for this case and send a PR then.
One thing that I discovered while were debugging is that this is reproduces for constructors in compiled code only while it works properly for types defined in the same project.

**Comment by dsyme** [comment]
@auduchinok Thanks, that would be very helpful

**Comment by auduchinok** [comment]
I couldn't reproduce it in tests, but managed to fix both issues in #777.
